### ISSUE TITLE
Update references to the renamed symbol in the entire project

### ISSUE THREAD

**Issue by jakubkaczor** [Myriad-Dreamin/tinymist#446]
Renaming a symbol updates only the references in the file it is defined in. When the item is imported in some other file, the references get broken. I expect them to be updated regardless of the place of reference, that is, throughout the entire project. For example:

```typst
// File: definition.typ
#let a = "a"
```

```typst
// File: reference.typ, in the same directory as definition.typ.
#import "./definition.typ": a

#a
```

Currently, renaming `a` in definition.typ breaks the reference in `reference.typ`. I would expect renaming `a` to `b` to result in
```typst
// File: definition.typ
#let b = "a"
```

```typst
// File: reference.typ, in the same directory as definition.typ.
#import "./definition.typ": b

#b
```

**Comment by Myriad-Dreamin** [comment]
I have tested it and find it work as expected, so I guess the `reference.typ` is not in your workspace. The workspace files are all of that are in your the opening *folder* recursively, but the following superfluous build directories are excluded.

https://github.com/Myriad-Dreamin/tinymist/blob/5b7794a5879930ed8e37ce9252c8adaf38731bc4/crates/tinymist-query/src/syntax/module.rs#L98-L108

**Comment by jakubkaczor** [comment]
That is most likely my mistake, sorry. I tested on a different example, and it appeared to me to be obvious for the feature to be partly missing (this was a feature request). Therefore, instead of giving a tested, minimal, reproducible example of a bug, I gave an example for the feature description. I am not sure what went wrong then.

I must have inclined for the missing feature, because I previously used typst-lsp that does not support renaming symbols and this is a younger project, for which I am thankful. 🙂
